ChiModules/LinearBoltzmannSolver/lbs_01d_initparrays.cc

```diff
@@ -86,79 +51,90 @@ void LinearBoltzmann::Solver::InitializeParrays()
   // we need to solve a matrix which square size is the
-  // amount of dofs on the cell. max_cell_dof_count is
+  // amount of nodes on the cell. max_cell_dof_count is
   // initialized here.
   //
-  int block_MG_counter = 0;       //Counts the strides of moment and group
+  size_t block_MG_counter = 0;       //Counts the strides of moment and group
 
   chi_mesh::Vector3 ihat(1.0, 0.0, 0.0);
   chi_mesh::Vector3 jhat(0.0, 1.0, 0.0);
   chi_mesh::Vector3 khat(0.0, 0.0, 1.0);
 
-  grid_nodal_mappings.reserve(grid->local_cells.size());
-  if (cell_transport_views.empty())
+  cell_transport_views.clear();
+  cell_transport_views.reserve(grid->local_cells.size());
+  for (auto& cell : grid->local_cells)
   {
-    for (auto& cell : grid->local_cells)
-    {
-      size_t num_nodes = discretization->GetCellNumNodes(cell);
-
-      CellLBSView cell_lbs_view(num_nodes, num_grps, M);
-
-      int mat_id = cell.material_id;
-
-      cell_lbs_view.xs_id = matid_to_xs_map[mat_id];
+    size_t num_nodes  = discretization->GetCellNumNodes(cell);
+    int    mat_id     = cell.material_id;
+    int    xs_mapping = matid_to_xs_map[mat_id];
 
-      cell_lbs_view.dof_phi_map_start = block_MG_counter;
-      block_MG_counter += num_nodes * num_grps * num_moments;
+    size_t cell_phi_address = block_MG_counter;
 
-      chi_mesh::sweep_management::CellFaceNodalMapping cell_nodal_mapping;
-      cell_nodal_mapping.reserve(cell.faces.size());
-
-      //Init face upwind flags and adj_partition_id
-      cell_lbs_view.face_local.resize(cell.faces.size(), true);
-      int f=0;
-      for (auto& face : cell.faces)
+    std::vector<bool> face_local_flags;
+    face_local_flags.resize(cell.faces.size(), true);
+    bool cell_on_boundary = false;
+    int f=0;
+    for (auto& face : cell.faces)
+    {
+      if (not face.has_neighbor)
       {
-        if (not face.has_neighbor)
-        {
-          chi_mesh::Vector3& n = face.normal;
-
-          int boundary_id = -1;
-          if      (n.Dot(ihat)>0.999)  boundary_id = 0;
-          else if (n.Dot(ihat)<-0.999) boundary_id = 1;
-          else if (n.Dot(jhat)> 0.999) boundary_id = 2;
-          else if (n.Dot(jhat)<-0.999) boundary_id = 3;
-          else if (n.Dot(khat)> 0.999) boundary_id = 4;
-          else if (n.Dot(khat)<-0.999) boundary_id = 5;
-
-          if (boundary_id >= 0) face.neighbor_id = boundary_id;
-        }//if bndry
-
-        if (not face.IsNeighborLocal(*grid))
-          cell_lbs_view.face_local[f] = false;
-
-        //Local nodal mappings
-        std::vector<short> face_nodal_mapping;
-        int ass_face = -1;
-
-        if (face.has_neighbor and face.IsNeighborLocal(*grid))
-        {
-          grid->FindAssociatedVertices(face,face_nodal_mapping);
-          ass_face = face.GetNeighborAssociatedFace(*grid);
-        }
+        chi_mesh::Vector3& n = face.normal;
+
+        int boundary_id = -1;
+        if      (n.Dot(ihat)>0.999)  boundary_id = 0;
```

Comment:
is 0.999 good enough? This has me worried. More digits?

Reply:
Nope. This is a hold-over from requiring the grid over dimension to have xmin,xmax,ymin,etc boundaries. If you would allow this one to pass, I have a big boundary update planned that will cover checking quadratures and boundaries for reflection specifications.

The way SPDS is set up now allows for arbitrary boundaries so boundary-type specifications need a lot of improvement.